src/ondisk.rs

```diff
@@ -8328,15 +8900,71 @@ make_token_accessors! {
         DfRemapAt1TiB(default 0, id 0x35ee_96f3) | pub get DfRemapAt1TiB : pub set DfRemapAt1TiB,
         DfXgmiTxEqMode(default 0xff, id 0xade7_9549) | pub get DfXgmiTxEqMode : pub set DfXgmiTxEqMode,
         DfInvertDramMap(default 0, id 0x6574_b2c0) | pub get DfToggle : pub set DfToggle,
+        DfXgmiCrcScale(default 5, id 0x5174_f4a0) | pub get u8 : pub set u8,
+        DfCxlMemInterleaving(default 0xFF, id 0x6387_09e6) | pub get DfCxlToggle : pub set DfCxlToggle,
+        DfCxlSublinkInterleaving(default 0xFF, id 0x66f_1625a) | pub get DfCxlToggle : pub set DfCxlToggle,
+        // TODO: Represent bitfield somehow
+        #[cfg_attr(feature = "serde-hex", serde(serialize_with = "SerHex::<StrictPfx>::serialize", deserialize_with = "SerHex::<StrictPfx>::deserialize"))]
+        DfXgmiAcDcCoupledLink(default 0xff, id 0xa7ae_5713) | pub get u8 : pub set u8,
```

Comment:
Sure, why not? modular_bitfield should work. See MemPmuBistTestSelect for how. For example:

```
        #[derive(BitfieldSpecifier, Debug, PartialEq)]
        #[bits = 1]
        pub enum DfXgmiAcDcMode {
            AcCoupled = 0,
            DcCoupled = 1,
        }

make_bitfield_serde! {
    #[bitfield(bits = 8)]
    #[repr(u8)]
    #[derive(Debug, Copy, Clone, PartialEq)]
    pub struct DfXgmiAcDcCoupledLink {
        pub socket_0_link_0 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
        pub socket_0_link_1 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
        pub socket_0_link_2 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
        pub socket_0_link_3 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
        pub socket_1_link_0 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
        pub socket_1_link_1 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
        pub socket_1_link_2 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
        pub socket_1_link_3 || #[serde(default)] bool : bool | pub get DfXgmiAcDcMode : pub set DfXgmiAcDcMode,
    }
}
impl_bitfield_primitive_conversion!(DfXgmiAcDcCoupledLink, 0b1111_1111, u8);

```

Or maybe a per-socket struct with bitfieldspecfier, see Ddr4DimmRanks for how.

I can also add it myself if you want.

Reply:
Not sure how many explicit arrays we want here. One?

Reply:
Updated to use bitfield